types-core/index.d.ts

```diff
@@ -83,6 +89,21 @@ declare module '@azure/functions-core' {
         error: any;
     }
 
+    /**
+     * Context on a function app that is about to be started
+     * This object will be passed to all app start hooks
+     */
+    export interface AppStartContext extends HookContext {
+        /**
+         * Absolute directory of the function app
+         */
+        functionAppDirectory: string;
+        /**
+         * The version of the host running the function app
+         */
+        hostVersion?: string;
```

Comment:
Any idea why this would ever be undefined? It would be more convenient for users if it was not optional.

Reply:
Looking at host code, it shouldn't ever be `undefined`, so I think we should be able to use `nonNullProp` here, but the same could be said about `functionAppDirectory`. Ideally we should treat them similarly?

Reply:
The host version is a static string that has never and will never change during the lifecycle of a worker. We already know one scenario where the functionAppDirectory changes values during the worker's lifecycle (specialization) and I could see more scenarios popping up in the future. The difference is that if `hostVersion` is undefined, that's a bug in the host and we should fix that. If `functionAppDirectory` is undefined, that's reasonable behavior

Reply:
Ok, I changed it to `nonNullProp`